### PR TITLE
Coin

### PR DESCRIPTION
**Pull request by looooo**
new pull request for coin3d.

### CODE REVIEW

**Comment by looooo** [comment]
i think this is ready now

File: recipes/coin/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  track_features:
+    - vc14  # [win and py35]
+
+requirements:
+  build:
+    - cmake  # [unix]
+    - ninja  # [win]
+    - python  # [win]
+    - boost 1.61.*
+    - toolchain  # [osx]
```
Comment (jakirkham) [inline recipes/coin/meta.yaml]: No need for a selector here.

File: recipes/coin/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    - test.sh  # [osx]
+  commands:
+    - g++ -o test test.cpp -lCoin -L${PREFIX}/lib/ -I${PREFIX}/include/  # [linux]
+    - ./test  # [linux]
+    - bash test.sh  # [osx]
+    - where /r "%PREFIX%"  Coin4.dll  # [win]
+
+about:
+  home: https://bitbucket.org/Coin3D/coin/wiki/Home
+  license: BSD 3-Clause License
```
Comment (jakirkham) [inline recipes/coin/meta.yaml]: `BSD 3-Clause` is fine. Please drop `License`.

Comment (jakirkham) [inline recipes/coin/meta.yaml]: Could optionally add `license_family: BSD`.

Comment (jakirkham) [inline recipes/coin/meta.yaml]: Is there a license file we can include?

File: recipes/coin/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  summary: this is coin3d
```
Comment (jakirkham) [inline recipes/coin/meta.yaml]: Maybe something a bit more descriptive. 😉

File: recipes/coin/test.cpp [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#include <Inventor/nodes/SoSeparator.h>
+#include <iostream>
+
+int main(int argc, char** argv)
+{
+    std::cout << "running a little test" << std::endl;
+    SoSeparator* my_node;
+    return 0;
+}
```
Comment (jakirkham) [inline recipes/coin/test.cpp]: nit: Please add a terminal newline.

File: recipes/coin/test.sh [hunk 5]
```diff
@@ -0,0 +1 @@
+[ -f ${PREFIX}/lib/libCoin.dylib ] && exit 0 || exit 1
```
Comment (jakirkham) [inline recipes/coin/test.sh]: nit: Please add a terminal newline.

File: recipes/coin/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  run:
+    - boost 1.61.*
+
+
+test:
+  files:
+    - test.cpp  # [linux]
+    - test.sh  # [osx]
+  commands:
+    - g++ -o test test.cpp -lCoin -L${PREFIX}/lib/ -I${PREFIX}/include/  # [linux]
```
Comment (jakirkham) [inline recipes/coin/meta.yaml]: Can we test this with `clang++` on Mac?

File: recipes/coin/test.sh (hunk 5 above)
Comment (jakirkham) [inline recipes/coin/test.sh]: Also this can just be a `test -f` statement in the `test/commands` section.

File: recipes/coin/meta.yaml (hunk 6 above)
Comment (looooo) [inline recipes/coin/meta.yaml]: I tried it, but failed with linking... The error was something about libCoin.dynlib not in @rpath.

File: recipes/coin/test.sh (hunk 5 above)
Comment (looooo) [inline recipes/coin/test.sh]: this way?
- test -f ${PREFIX}/lib/libCoin.dylib  # [osx]

File: recipes/coin/meta.yaml (hunk 2 above)
Comment (looooo) [inline recipes/coin/meta.yaml]: I haven't found one. Should I include a copy of BSD 3 license in the recipe-dir?

File: recipes/coin/test.sh (hunk 5 above)
Comment (jakirkham) [inline recipes/coin/test.sh]: Yeah, that would work fine.

**Comment by jakirkham** [comment]
Toggling to retest linter.

**Comment by jakirkham** [comment]
Apparently we were missing AppVeyor as well.

**Comment by looooo** [comment]
no idea why appveyor wasn't building the last time. But I think now everything should be ready.

**Comment by looooo** [comment]
@jakirkham I think this is ready to be merged/created/feed-stocked ;)

File: recipes/coin/test.cpp (hunk 4 above)
Comment (jakirkham) [inline recipes/coin/test.cpp]: Done.

File: recipes/coin/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,57 @@
+package:
+  name: coin
+  version: 4.0.0
+
+source:
+  fn: coin.tar.gz
+  url: https://bitbucket.org/Coin3D/coin/get/CMake.tar.gz
```
Comment (jakirkham) [inline recipes/coin/meta.yaml]: Is there a more stable URL we can use?

File: recipes/coin/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+  commands:
+    - g++ -o test test.cpp -lCoin -L${PREFIX}/lib/ -I${PREFIX}/include/  # [linux]
+    - ./test  # [linux]3
+    - test -f ${PREFIX}/lib/libCoin.dylib  # [osx]
+    - where /r "%PREFIX%"  Coin4.dll  # [win]
+
+about:
+  home: https://bitbucket.org/Coin3D/coin/wiki/Home
+  license: BSD 3-Clause
+  license_family: BSD
```
Comment (jakirkham) [inline recipes/coin/meta.yaml]: Please add `license_file: COPYING` to package the license.

**Comment by jakirkham** [comment]
Sorry for the super long turnaround. Couple minor things noted above.

**Comment by jakirkham** [comment]
Could you please also take a look at this one, @scopatz, when you have the chance?

File: recipes/coin/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+
+test:
+  requires:
+    - python {{ environ['PY_VER'] + '*' }}  # [win]
+  files:
+    - test.cpp  # [linux]
+  commands:
+    - g++ -o test test.cpp -lCoin -L${PREFIX}/lib/ -I${PREFIX}/include/  # [linux]
+    - ./test  # [linux]3
```
Comment (scopatz) [inline recipes/coin/meta.yaml]: Why the 3 afterwards?

File: recipes/coin/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,57 @@
+package:
+  name: coin
```
Comment (scopatz) [inline recipes/coin/meta.yaml]: Maybe this should be called coin3d, to distinguish it from other coins.

**Comment by scopatz** [comment]
I agree with your suggestions @jakirkham. To be clear though, I have no special knowledge of this Coin, which seems to be a graphics engine, as opposed to Coin the linear programming and mixed integer solver that I am familiar with.

File: recipes/coin/meta.yaml (hunk 7 above)
Comment (looooo) [inline recipes/coin/meta.yaml]: I don't think there is. This cmake-branch is quite new, but simplify the build process a lot. Maybe it's possible to use the latest commit as url?

**Comment by looooo** [comment]
sorry, for the misleading naming. This library is quite old and is named libcoin80 in ubuntu. But to make things clearer I will rename it. Never heard of any other coin...

**Comment by looooo** [comment]
i would like to use this library on windows only with vc14. Is it ok to skip python2.7.
BTW. Wouldn't it be possible to create an empty vc14 which introduce the vc14 feature. This way we could track this feature without the need to have python as dependency.

File: recipes/coin/meta.yaml (hunk 7 above)
Comment (looooo) [inline recipes/coin/meta.yaml]: What about making a tag with the zipped source once the pr is accepted and the repository is created?

**Comment by jakirkham** [comment]
Hi @looooo, sorry we haven't been very attentive. Just very busy. Is there a reason you closed this? Do you still need it? Something we can help with?

**Comment by looooo** [comment]
thanks for the help. As we need some time to get everything running it is the better idea to first try to get the recipes working outside of conda-forge. We have a repository for all the needed recipes: https://github.com/looooo/FreeCAD_Conda
I will come back to conda-forge once everything is stable enough. But this will need some time :-)

**Pull request closed without merging**